### ISSUE TITLE
UnusedPrivateMember rule gets triggered on a loop range

### ISSUE THREAD

**Issue by gusega** [detekt/detekt#4879]
## Observed Behavior

The following code:
```
            for (i in 0..5) {
                doSomething()
            }

```
triggers the following check: 
Private property `i` is unused. [UnusedPrivateMember]

## Expected Behavior
As there is no way In kotlin to do a range loop without the variable, then such a code should cause the error being reported.

## Your Environment
* Detekt maven plugin version: 1.20.0
* Detekt version: 1.20.0
